FTBFS on mantic / noble on ppc64el

Basically the same as #309, but only on ppc64el:

```
In function ‘_Construct’,
    inlined from ‘_M_construct’ at /usr/include/c++/13/optional:278:19,
    inlined from ‘_M_move_assign’ at /usr/include/c++/13/optional:198:26,
    inlined from ‘operator=’ at /usr/include/c++/13/optional:420:22,
    inlined from ‘operator=’ at /usr/include/c++/13/optional:670:23,
    inlined from ‘operator=’ at /usr/include/c++/13/optional:705:11,
    inlined from ‘operator=’ at /usr/src/wlcs-1.6.1+dev5-g407c00a-0ubuntu24.04/include/in_process_server.h:216:8,
    inlined from ‘done_thunk’ at /usr/src/wlcs-1.6.1+dev5-g407c00a-0ubuntu24.04/src/in_process_server.cpp:1144:57:
/usr/include/c++/13/bits/stl_construct.h:119:7: error: ‘MEM[(struct pair &)&D.14449 + 8]’ may be used uninitialized [-Werror=maybe-uninitialized]
  119 |       ::new((void*)__p) _Tp(std::forward<_Args>(__args)...);
      |       ^
```

E.g. https://launchpad.net/~mir-team/+archive/ubuntu/dev/+build/26994130

--- a/debian/rules
+++ b/debian/rules
@@ -26,7 +26,7 @@ endif
 
 # https://gcc.gnu.org/bugzilla/show_bug.cgi?id=105329
 ifeq ($(DEB_HOST_ARCH), ppc64el)
-  ifneq ($(shell gcc --version | grep '12.[[:digit:]]\+.[[:digit:]]\+$$'),)
+  ifneq ($(shell gcc --version | grep '1[23].[[:digit:]]\+.[[:digit:]]\+$$'),)
     export DEB_CFLAGS_MAINT_APPEND = -O2
     export DEB_CXXFLAGS_MAINT_APPEND = -O2
     export DEB_FCFLAGS_MAINT_APPEND = -O2

